packages/@uppy/xhr-upload/src/index.ts

```diff
@@ -218,7 +219,15 @@ export default class XHRUpload<
             },
           })
 
-          const body = JSON.parse(res.responseText) as B
+          let body = await this.opts.getResponseData?.(res)
+          try {
+            body ??= JSON.parse(res.responseText) as B
```

Comment:
Sorry for interfering.

This code overrides the value returned by `getResponseData`. For me, as a user, this behavior would be unexpected. The `responseText` might be a proper JSON document but with wrong field names that I would like to modify in `getResponseData`. I propose to check whether `getResponseData` is provided before parsing the response.

Reply:
That's not what happens here. Through optional chaining, `getResponseData` is only called if it exists, otherwise `body` is `undefined`. `JSON.parse` is only called if `body` is `undefined` (`??=`).

Reply:
ah, overlooked this 😄 well done!